core/domain/suggestion_services_test.py

```diff
@@ -2084,6 +2094,1271 @@ def test_create_translation_contribution_stats_from_model(self) -> None:
+        self.assertEqual(
+            translation_contribution_stats_model.accepted_translations_count,
+            2
+        )
+        self.assertEqual(
+            (
+                translation_contribution_stats_model
+                .accepted_translations_without_reviewer_edits_count
+            ),
+            1
+        )
+
+    def test_update_question_contribution_stats_when_submitting(self) -> None:
+        # Steps required in the setup phase before testing.
+        # 1. Save new skills.
+        # 2. Save a topic assigning skills for it.
+        # 4. Create a question suggestion.
+        skill_id_1 = skill_services.get_new_skill_id()
+        skill_id_2 = skill_services.get_new_skill_id()
+        self.save_new_skill(
+            skill_id_1, self.author_id, description='description')
+        self.save_new_skill(
+            skill_id_2, self.author_id, description='description')
+        topic_id = topic_fetchers.get_new_topic_id()
+        self.save_new_topic(
+            topic_id, 'topic_admin', name='Topic1',
+            abbreviated_name='topic-three', url_fragment='topic-three',
+            description='Description',
+            canonical_story_ids=[],
+            additional_story_ids=[],
+            uncategorized_skill_ids=[skill_id_1, skill_id_2],
+            subtopics=[], next_subtopic_id=1)
+        suggestion_change_1: Dict[
+            str,
+            Union[str, float, Dict[str, Union[
+                str, List[str], int, state_domain.StateDict]]]] = {
+            'cmd': (
+                question_domain
+                .CMD_CREATE_NEW_FULLY_SPECIFIED_QUESTION),
+            'question_dict': {
+                'question_state_data': self._create_valid_question_data(
+                    'default_state').to_dict(),
+                'language_code': 'en',
+                'question_state_data_schema_version': (
+                    feconf.CURRENT_STATE_SCHEMA_VERSION),
+                'linked_skill_ids': ['skill_1'],
+                'inapplicable_skill_misconception_ids': ['skillid12345-1']
+            },
+            'skill_id': skill_id_1,
+            'skill_difficulty': 0.3
+        }
+        suggestion_change_2: Dict[
+            str,
+            Union[str, float, Dict[str, Union[
+                str, List[str], int, state_domain.StateDict]]]] = {
+            'cmd': (
+                question_domain
+                .CMD_CREATE_NEW_FULLY_SPECIFIED_QUESTION),
+            'question_dict': {
+                'question_state_data': self._create_valid_question_data(
+                    'default_state').to_dict(),
+                'language_code': 'en',
+                'question_state_data_schema_version': (
+                    feconf.CURRENT_STATE_SCHEMA_VERSION),
+                'linked_skill_ids': ['skill_2'],
+                'inapplicable_skill_misconception_ids': ['skillid12345-1']
+            },
+            'skill_id': skill_id_2,
+            'skill_difficulty': 0.3
+        }
```

Comment:
Probably also want helper functions to create the question suggestions, so that these lines can become something like `create_question_suggestion(skill_id)`. Ditto below.

Reply:
Done